crossdock/docker-compose.yml

```diff
@@ -3,6 +3,7 @@ version: '2'
 services:
     crossdock:
         image: crossdock/crossdock
+        dns_search: .
```

Comment:
I have attempted this change in #1029. All crossdock examples show this `dns_search` line. The reasoning is unclear. Alpine has/had DNS issues that might explain this. It definitely helped me locally when crossdock failed to connect to `go:8080`. Here is the commit from crossdock related to this line: https://github.com/crossdock/crossdock/pull/64. Please let me know if you are strongly opposed to adding this.

Reply:
what does it do?

Reply:
From this golang issue (https://github.com/golang/go/issues/15419), it keeps Go from trying to resolve local container names to external URLs. Here is a brief quote from that issue:

> The way how pure Go DNS resolver resolves names caused an error in my program. When I set "ndots:" to 0 and resolved "mysql", I wanted to resolve "mysql." first, but instead, pure Go DNS resolver resolved "mysql.searchdomain." first. As it happened, the domain name"mysql.searchdomain." was valid in my DNS server, so I got a wrong IP of mysql.

Reply:
thanks - sure, let's keep it. I am still not clear where this PR is going, since it has too many things (and the shared base image with Cassandra included is a non-starter).

Reply:
Well the PR overall is to add a true "wait for Cassandra" to the crossdock tests. For months, I have found crossdock to be flakey. Now gRPC made it worse. I can split the fixes out into separate pull requests, but I believe fixing these issues are worthwhile.

Regarding the use of a base image, I'm not sure why we didn't do this in the first place. It just simplifies building two nearly identical images (jaeger-collector and jaeger-query) from a standard source. The reason for using Cassandra is that it is the only reliable way to guarantee Cassandra is ready for operation.

An alternative approach would be using a memory store like we've discussed for a while, but that hasn't happened yet.

Reply:
* I am fine with "wait for cassandra" approach for crossdock
  * longer term we should avoid using Cassandra for crossdock in the first place (#1221)
* I am **not** fine with including Cassandra dependency into the default (production) images
* I am fine with using a shared base image, as long as it's not going to create more friction, i.e. if it can be completely local during the build and not pushed to Docker Hub. But that base image should have no traces of Cassandra.

Between "wait for Cassandra", using "dns_search", and defining base image, there are at least 3 independent PRs.

Reply:
Right I think avoiding Cassandra is the simplest way to fix most of these issues (DNS stuff still needs another fix if it comes up again).